Components/Sources/Components/Components/Watchlist/WKWatchlistView.swift

```diff
@@ -3,84 +3,95 @@ import WKData
 
 // MARK: - WKWatchlistView
 struct WKWatchlistView: View {
-
-	@ObservedObject var appEnvironment = WKAppEnvironment.current
- 	@ObservedObject var viewModel: WKWatchlistViewModel
+    
+    @ObservedObject var appEnvironment = WKAppEnvironment.current
+    @ObservedObject var viewModel: WKWatchlistViewModel
     var emptyViewModel: WKEmptyViewModel
-	weak var delegate: WKWatchlistDelegate?
-    weak var loggingDelegate: WKWatchlistLoggingDelegate?
+    weak var delegate: WKWatchlistDelegate?
+	weak var loggingDelegate: WKWatchlistLoggingDelegate?
     weak var emptyViewDelegate: WKEmptyViewDelegate? = nil
-	weak var menuButtonDelegate: WKMenuButtonDelegate?
-
-	// MARK: - Lifecycle
-
-	var body: some View {
-		ZStack {
-			Color(appEnvironment.theme.paperBackground)
-				.ignoresSafeArea()
-			contentView
-		}.onAppear {
-            viewModel.fetchWatchlist {
+    weak var menuButtonDelegate: WKMenuButtonDelegate?
+    
+    // MARK: - Lifecycle
+    
+    var body: some View {
+        ZStack {
+            Color(appEnvironment.theme.paperBackground)
+                .ignoresSafeArea()
+            contentView
+        }.onAppear {
+           viewModel.fetchWatchlist {
                 let items = viewModel.sections.flatMap { $0.items }
                 loggingDelegate?.logWatchlistDidLoad(itemCount: items.count)
             }
-		}
-	}
-
-	@ViewBuilder
-	var contentView: some View {
-		if viewModel.hasPerformedInitialFetch {
-			if viewModel.sections.count > 0 {
-				WKWatchlistContentView(viewModel: viewModel, delegate: delegate, menuButtonDelegate: menuButtonDelegate)
-			} else if viewModel.sections.count == 0 && viewModel.activeFilterCount > 0 {
-				WKEmptyView(viewModel: emptyViewModel, delegate: emptyViewDelegate, type: .filter)
-			} else {
-				WKEmptyView(viewModel: emptyViewModel, delegate: emptyViewDelegate, type: .noItems)
-			}
-		} else {
-			ProgressView()
-		}
-	}
-
+        }
+    }
+    
+    @ViewBuilder
+    var contentView: some View {
+        if viewModel.hasPerformedInitialFetch {
+            if viewModel.sections.count > 0 {
+                WKWatchlistContentView(viewModel: viewModel, delegate: delegate, menuButtonDelegate: menuButtonDelegate)
+            } else if viewModel.sections.count == 0 && viewModel.activeFilterCount > 0 {
+                WKEmptyView(viewModel: emptyViewModel, delegate: emptyViewDelegate, type: .filter)
+            } else {
+                WKEmptyView(viewModel: emptyViewModel, delegate: emptyViewDelegate, type: .noItems)
+            }
+        } else {
+            ProgressView()
+            .accessibilityHidden(true)
```

Comment:
Should we retain the intrinsic accessibility nature of this view here? Using the modifier means the user gets trapped with no indication about the view itself as it's loading, which could be especially problematic if it's taking a long time.